packages/insomnia-scripting-environment/src/objects/send-request.ts

```
@@ -30,20 +30,32 @@ export async function sendRequest(
           return curlOutputToResponse(output, request);
         })
         .then((transformedOutput: Response) => {
-          cb(undefined, transformedOutput);
+          if (cb) {
+            cb(undefined, transformedOutput);
+          }
           resolve(transformedOutput);
         })
         .catch((e: string | undefined) => {
-          cb(e, undefined);
-          resolve(undefined);
+          if (cb) {
+            cb(e, undefined);
+          } else {
+            reject(e);
+          }
         });
     } catch (err: any) {
       if (err.name === 'AbortError') {
-        cb(`Request was cancelled: ${err.message}`, undefined);
+        if (cb) {
```

🤔 From the interface `cb` always exists, is it necessary to check? Or is this only for the script, so users may not pass in cb when they write the script?

---

Good point, as our implementation is in ts while users are writing script in js, it just tries to avoid swallowing error if no function is there.